atlasdb-feign/src/main/java/com/palantir/atlasdb/http/v2/ConjureJavaRuntimeTargetFactory.java

```diff
@@ -0,0 +1,115 @@
+import com.palantir.conjure.java.config.ssl.TrustContext;
+import com.palantir.conjure.java.ext.refresh.Refreshable;
+import com.palantir.conjure.java.okhttp.HostMetricsRegistry;
+
+public final class ConjureJavaRuntimeTargetFactory implements TargetFactory {
+    private static final HostMetricsRegistry HOST_METRICS_REGISTRY = new HostMetricsRegistry();
+
+    public static final ConjureJavaRuntimeTargetFactory DEFAULT = new ConjureJavaRuntimeTargetFactory();
+    public static final String CLIENT_VERSION_STRING = "Conjure-Java-Runtime";
+
+    private ConjureJavaRuntimeTargetFactory() {
+        // Use the instances.
+    }
+
+    @Override
+    public <T> InstanceAndVersion<T> createProxy(Optional<TrustContext> trustContext, String uri, Class<T> type,
+            AuxiliaryRemotingParameters parameters) {
+        ClientConfiguration clientConfiguration = ClientOptions.DEFAULT_RETRYING.create(
+                ImmutableList.of(uri),
+                Optional.empty(),
+                trustContext.orElseThrow(() -> new IllegalStateException("CJR requires a trust context")));
+        return wrapWithVersion(JaxRsClient.create(
+                type,
+                addAtlasDbRemotingAgent(parameters.userAgent()),
+                HOST_METRICS_REGISTRY,
+                clientConfiguration));
+    }
+
+    @Override
+    public <T> InstanceAndVersion<T> createProxyWithFailover(
+            ServerListConfig serverListConfig,
+            Class<T> type,
+            AuxiliaryRemotingParameters parameters) {
+        ClientConfiguration clientConfiguration = ClientOptions.DEFAULT_RETRYING.serverListToClient(serverListConfig);
+
+        return wrapWithVersion(JaxRsClient.create(
+                type,
+                addAtlasDbRemotingAgent(parameters.userAgent()),
+                HOST_METRICS_REGISTRY,
+                clientConfiguration));
+    }
+
+    @Override
+    public <T> InstanceAndVersion<T> createLiveReloadingProxyWithFailover(
+            Supplier<ServerListConfig> serverListConfigSupplier,
+            Class<T> type,
+            AuxiliaryRemotingParameters parameters) {
+        Supplier<ServerListConfig> nonEmptyServerList = () -> injectDummyServer(serverListConfigSupplier);
+
+        // TODO (jkong): Thread leak for days. Though no regression over existing implementation in Feign.
+        Refreshable<ClientConfiguration> refreshableConfig = PollingRefreshable
+                .createComposed(nonEmptyServerList,
+                        Duration.ofSeconds(5L),
+                        ClientOptions.DEFAULT_RETRYING::serverListToClient)
+                .getRefreshable();
+        return wrapWithVersion(JaxRsClient.create(
+                type,
+                addAtlasDbRemotingAgent(parameters.userAgent()),
+                HOST_METRICS_REGISTRY,
+                refreshableConfig));
+    }
+
+    // TODO (gmaretic): This is a hack because CJR doesn't like configurations with 0 servers, yet we claim
```

Comment:
if CJR doesn't like configurations with 0 servers, then we have to reassess whether or not we're realistically getting these configurations in k8s, would rather ask and then delete rather than keep since we'll never actually look at this again tbh.

Reply:
@ashrayjain I recall the rationale for doing this in the past was to allow timelock to bootstrap itself in k8s - could we check if this is still correct / needed?